runtimerevolution/web3-roulette
# [WEB3-39] Api authentication

# Description

This PR adds authentication to the API. It takes advantage of the google auth to validate the user on the API side. Two new endpoints were added: /login and /me. One to authenticate using the access token of google authentication, and the other to access your user information.

Ticket: https://runtime-revolution.atlassian.net/browse/WEB3-39

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Try to use the endpoints without the token and verify that you don't have access
- Try to use /login endpoint to get a token and /me to get user information
- Verify that with that token you have access to other endpoints
- New tests were added to ensure the functionality

## Checklist:
